### ISSUE TITLE
Problems encountered during training

### ISSUE THREAD

**Issue by lrnb12345** [ZhaochongAn/COSeg#12]
Hello, after pre training on my own dataset, I conducted a small sample 1way1shot training, but encountered an accuracy of 0 during this process. I would like to ask if this is normal.
 Pyhon3 main_f.spy -- config/s3disi_CSeg_fs.yaml savepath/root/autodl-tmp/COSeg-miin/1w1s_train pretrain_mackbone/root/autodl-tmp/COSeg-miin/prtrain cvfold 0 n_way 1 k_shot 1 num_ episode per_comb 1000. This is my instruction.
![屏幕截图 2024-08-22 161337](https://github.com/user-attachments/assets/6e22dbdb-5df6-43d1-9451-e80724517ace)

**Comment by ZhaochongAn** [comment]
Hi, it seems not normal. Does the loss decrease during episodic training? What is the pre-training performance? If you are using new dataset, it is also necessary to make sure the input points and labels are correct.

**Comment by lrnb12345** [comment]
> 你好，这似乎不正常。在情景训练期间损失会减少吗？预训练性能如何？如果您使用的是新数据集，还需要确保输入点和标签正确。


306/10000
实时翻译
在costg.py文件中，我的数据集是基于sdis数据集创建的，除“标志”外共有九个类别。我注意到，在s3dis中，除了“旗帜”之外，还有12个类别，每个类别都有6个类别。然而，我只有9个类别，不能在每个文件夹中归入同一类别。我该怎么办

划译
In the costg.py file, my dataset was created based on the sdis dataset, with a total of nine categories excluding the 'flags'. I noticed that in s3dis, there were 12 categories excluding the' flags' and six categories were set in each fold. However, I only have 9 categories and cannot be classified into the same category in each fold. What should I do，
In the costg.py file, my dataset was created based on the sdis dataset, with a total of nine categories excluding the 'flags'. I noticed that in s3dis, there were 12 categories excluding the' flags' and six categories were set in each fold. However, I only have 9 categories and cannot be classified into the same category in each fold. What should I do
Also, is there a correlation between setting the fold partition in sdis_f.spy and setting it in coseg.py
![Uploading 屏幕截图 2024-08-22 215654.png…]()

![屏幕截图 2024-08-22 213901](https://github.com/user-attachments/assets/cff82f9e-9a1b-48f7-bb9c-043324d3529f)

**Comment by ZhaochongAn** [comment]
"9 categories"

The category fold division should be fine. 

"Also, is there a correlation between setting the fold partition in s3dis_fs.py and setting it in coseg.py? "

You should change the base_class_to_pred_label mapping in coseg.py according to your category division defined in the your dataset. From what you posted here, I could see your base_class_to_pred_label is not changed in terms of your category division. That would lead to incorrect behavier of the base prototype learning.

**Comment by lrnb12345** [comment]
> "9 categories"
> 
> The category fold division should be fine.
> 
> "Also, is there a correlation between setting the fold partition in s3dis_fs.py and setting it in coseg.py? "
> 
> You should change the base_class_to_pred_label mapping in coseg.py according to your category division defined in the your dataset. From what you posted here, I could see your base_class_to_pred_label is not changed in terms of your category division. That would lead to incorrect behavier of the base prototype learning.

Thank you for your reply. Am I doing the right thing? Because the nine categories cannot be evenly divided into six categories per fold according to the s3dis dataset. If I modify my categories according to the s3dis_f.spy file, I will get an error message
![Uploading 屏幕截图 2024-08-24 115852.png…]()

**Comment by ZhaochongAn** [comment]
It seems that your dataset has 9 categories in total and you are using all the 9 categories as base classes? You should divide the 9 categories to two folds, such as one fold with 4 categories and the other with 5 categories. Please refer to the dataset file on S3DIS and ScanNet.

**Comment by lrnb12345** [comment]
> It seems that your dataset has 9 categories in total and you are using all the 9 categories as base classes? You should divide the 9 categories to two folds, such as one fold with 4 categories and the other with 5 categories. Please refer to our dataset definition file of S3DIS and ScanNet.

Thank you for your patient reply. After pre training the backbone, I followed your guidance to conduct few shot training. During the process, the program progress was as shown in the figure, with an accuracy of 1. I would like to ask if this is normal,
![屏幕截图 2024-08-27 112720](https://github.com/user-attachments/assets/e8dec8d2-0f41-4dc1-84c5-10df1d9aec6a)

**Comment by ZhaochongAn** [comment]
Hi, I think it is normal. In my experience, it's common to see high accuracy, which indicates that the model's predictions often encompass the ground truth foreground areas, and usually extend beyond them, as reflected by the relatively low IoU metrics. Also note that the accuracy metric is calculated only for the foreground classes.

**Comment by lrnb12345** [comment]
> I think it is normal. In my experience, it's common to see high accuracy, which indicates that the model's predictions often encompass the ground truth foreground areas, and usually extend beyond them, as reflected by the relatively low IoU metrics. Also note that the accuracy metric is calculated only for the foreground classes.

Thank you very much for your patient reply. I have successfully trained and tested to achieve the expected results. However, when I visualized the file generated from the source code (query.npy querylb. npy sup. npy suplb. npy pred. npy), only a small part of the scene appeared, instead of visualizing the entire room like the s3dis dataset. Is it because my data scene is too large? If so, how can I modify the code or related parameters to make the final visualization result a complete visualization of each scene, rather than a small part segmented from a scene.
I ran visualizing. py and the result is shown in the following figure, only a small part of my building scene appeared
![屏幕截图 2024-08-29 124346](https://github.com/user-attachments/assets/a2886f2f-12ec-4a1a-aea6-fb104bb6e61e)

**Comment by ZhaochongAn** [comment]
Hi, the figure seems a bit strange. Do you follow the instructions to run the visualization (https://github.com/ZhaochongAn/COSeg/tree/main?tab=readme-ov-file#visualization)? During the visualization, we crop the input point cloud if it have too many points and simply choose the quadrant with the most labelled points to avoid OOM issue. If you have enough GPU memory, you could adjust the point threshold to start the crop and the crop function as well. 

https://github.com/ZhaochongAn/COSeg/blob/806b2940d299ed551402293af42af0816ecbbc4e/util/s3dis_fs.py#L587-L595

https://github.com/ZhaochongAn/COSeg/blob/806b2940d299ed551402293af42af0816ecbbc4e/util/s3dis_fs.py#L534-L578
